src/OptionT.ts

```diff
@@ -1,21 +1,39 @@
-import { getSome } from './option/some';
-import { getNone } from './option/none';
 
-/**
- * An interface describing the argument passed to [[OptT]]'s `match` function.
- */
 export interface OptMatch<T, U, V> {
   some: (val: T) => U;
   none: () => V;
 }
 
+
 /**
- * An interface describing the `OptionT` type.
- *
- * Items of this type can either be a `Some` value (with something of
- * type `T` in them), or a `None` value (with nothing in them).
+ * An interface describing the argument passed to [[OptT]]'s `match` function.
  */
-export interface OptT<T> {
+
+export default abstract class OptionT<T> {
+  constructor(value?: T) {
```

Comment:
does this need to even have an argument at all?  Since you can't actually call it I'd think it could just be `constructor() {}`.

Reply:
I had originally left it, thinking we may want to do something with it if Some passed its value up through super, but I cant think of what that will be - so ill remove it.